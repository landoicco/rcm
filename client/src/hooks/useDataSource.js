//@flow

import { useEffect, useState } from "react";
import type { RCMDataEntity } from "../commons/types";

const useDataSource = (url: string): RCMDataEntity => {
  const initialData: RCMDataEntity = [];
  const [data, setData] = useState(initialData);

  useEffect(() => {
    fetch(url)
      .then((res) => res.json())
      .then((res) => setData(res));
  }, [url]);

  return data;
};

export default useDataSource;
