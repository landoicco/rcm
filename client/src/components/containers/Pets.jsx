//@flow

import React from "react";
import useDataSource from "../../hooks/useDataSource";
import { PETS_API_BASE_URL } from "../../commons/endpoints";
import type { Pet } from "../../commons/types";
import SinglePet from "../model/SinglePet";

const Pets = (): React.Node => {
  const pets: any = useDataSource(`${PETS_API_BASE_URL}/all`);
  return (
    <div className="container">
      <h3 className="center">Pets</h3>
      {pets.map((item: Pet) => (
        <SinglePet key={item.id} item={item} />
      ))}
    </div>
  );
};

export default Pets;
