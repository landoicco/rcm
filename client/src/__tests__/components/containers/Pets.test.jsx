import { render, screen } from "@testing-library/react";
import Pets from "../../../components/containers/Pets";
import useDataSource from "../../../hooks/useDataSource";

describe("Pets", () => {
  test("The component render correctly", () => {
    const { container } = render(<Pets />);
    expect(container).toMatchSnapshot();
  });

  test("Should have only one heading element", async () => {
    render(<Pets />);

    // Get all elements with a 'heading' role
    const elements = await screen.findAllByRole("heading");
    expect(elements.length === 1).toBe(true);
  });

  test("The word 'Pets' should be rendered at least once", async () => {
    render(<Pets />);

    // Find all elements with the word 'Pets' on it
    const elements = await screen.findAllByText("Pets");
    expect(elements.length > 0).toBe(true);
  });

  test("The useDataSource hook should be called one time", () => {});
});
