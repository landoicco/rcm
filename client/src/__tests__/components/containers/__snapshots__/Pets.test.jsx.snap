// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pets The component render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h3
      class="center"
    >
      Pets
    </h3>
  </div>
</div>
`;
